### PR TITLE
feat: add a new ratio config instead of table_data_deserialized_data_bytes

### PR DESCRIPTION
**Pull request by chienguo**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

- Fixes #[Link the issue here]
https://github.com/datafuselabs/databend/issues/14866

## Tests

- [ ] Unit Test
- [ ] Logic Test
- [ ] Benchmark Test
- [ ] No Test  - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/14896)

### CODE REVIEW

File: src/query/config/src/config.rs [hunk 1]
```diff
@@ -2803,18 +2803,18 @@ pub struct CacheConfig {
     #[serde(rename = "disk")]
     pub disk_cache_config: DiskCacheConfig,
 
-    /// Max size of in memory table column object cache. By default it is 0 (disabled)
+    /// Max percentage of in memory table column object cache relative to whole memory. By default it is 0 (disabled)
     ///
-    /// CAUTION: The cached items are deserialized table column objects, may take a lot of memory.
+    /// CAUTION: The cached items are deserialized table column objects, ma take a lot of memory.
```
Comment (dantengsky) [inline src/query/config/src/config.rs:2808]:
```suggestion
    /// CAUTION: The cached items are deserialized table column objects, may take a lot of memory.
```

File: src/query/config/src/inner.rs [hunk 2]
```diff
@@ -557,13 +557,13 @@ pub struct CacheConfig {
     /// Storage that hold the raw data caches
     pub disk_cache_config: DiskCacheConfig,
 
-    /// Max size of in memory table column object cache. By default it is 0 (disabled)
+    /// Max percentage of in memory table column object cache relative to whole memory. By default it is 0 (disabled)
     ///
-    /// CAUTION: The cache items are deserialized table column objects, may take a lot of memory.
+    /// CAUTION: The cached items are deserialized table column objects, ma take a lot of memory.
```
Comment (dantengsky) [inline src/query/config/src/inner.rs:562]:
```suggestion
    /// CAUTION: The cached items are deserialized table column objects, may take a lot of memory.
```

**Comment by dantengsky** [comment]
@chienguo  

Thanks for your contribution!

There's a concern that hoping you can consider in this PR:

In some scenarios (such as testing and deploying on cloud services), retaining the `table_data_deserialized_data_bytes` setting alongside `table_data_deserialized_memory_ratio` would be more convenient.

Furthermore, if both settings are non-zero, the `table_data_deserialized_data_bytes` setting should have higher priority.

Similar handling in the settings for spilling may serve as a reference (however, in the scenario of this PR, there's no need to consider settings at the perf_core level).

https://github.com/datafuselabs/databend/blob/345cba5aa73ba3e93f4c6372a75fb956812243e4/src/query/service/src/pipelines/builders/builder_sort.rs#L212C1-L239C5

cc @BohuTANG

**Comment by BohuTANG** [comment]
> @chienguo
> 
> Thanks for your contribution!
> 
> There's a concern that hoping you can consider in this PR:
> 
> In some scenarios (such as testing and deploying on cloud services), retaining the `table_data_deserialized_data_bytes` setting alongside `table_data_deserialized_memory_ratio` would be more convenient.
> 
> Furthermore, if both settings are non-zero, the `table_data_deserialized_data_bytes` setting should have higher priority.
> 
> Similar handling in the settings for spilling may serve as a reference (however, in the scenario of this PR, there's no need to consider settings at the perf_core level).
> 
> https://github.com/datafuselabs/databend/blob/345cba5aa73ba3e93f4c6372a75fb956812243e4/src/query/service/src/pipelines/builders/builder_sort.rs#L212C1-L239C5
> 
> cc @BohuTANG

@dantengsky Your suggestion looks great to me. 
@chienguo Sorry for issue https://github.com/datafuselabs/databend/issues/14866 , it seems it wasn't clear enough, @dantengsky added clarity.

**Comment by chienguo** [comment]
@dantengsky, already made the original one back, and have dealt with the priority

**Review by dantengsky — APPROVED** [review]

**Comment by dantengsky** [comment]
@chienguo  LGTM & thanks

**Comment by BohuTANG** [comment]
@chienguo 👍 . Thanks for your contribution.

**Comment by chienguo** [comment]
@BohuTANG, @dantengsky, thanks for your help

**Pull request merged**